ansible_builder/main.py

```diff
@@ -145,62 +148,67 @@ def get_dependency(self, entry):
-            CD = CollectionDefinition(path)
-            dep_file = CD.get_dependency(dep_type)
-            if not dep_file:
-                continue
-            namespace, name = CD.namespace_name()
-            ret.append(os.path.join(namespace, name, dep_file))
-        return ret
-
 
 class Containerfile:
     newline_char = '\n'
 
     def __init__(self, definition,
                  filename=constants.default_file,
                  build_context=constants.default_build_context,
-                 base_image=constants.default_base_image):
+                 base_image=constants.default_base_image,
+                 container_runtime=constants.default_container_runtime,
+                 tag=constants.default_tag):
 
         self.build_context = build_context
         os.makedirs(self.build_context, exist_ok=True)
         self.definition = definition
         self.path = os.path.join(self.build_context, filename)
         self.base_image = base_image
+        self.container_runtime = container_runtime
+        self.tag = tag
+        self.steps = []
 
-    def build_steps(self):
+    def prepare_galaxy_steps(self):
         self.steps = [
             "FROM {}".format(self.base_image),
             ""
         ]
-        requirements_path = self.definition.get_dependency('galaxy')
-        if requirements_path:
+        galaxy_requirements_path = self.definition.get_dependency('galaxy')
+        if galaxy_requirements_path:
+            # name is most likely "requirements.yml"
+            galaxy_requirements_name = os.path.basename(galaxy_requirements_path)
             # TODO: what if build context file exists? https://github.com/ansible/ansible-builder/issues/20
-            shutil.copy(requirements_path, self.build_context)
-            self.steps.extend(
-                GalaxySteps(
-                    os.path.basename(requirements_path)  # probably "requirements.yml"
-                )
-            )
+            dest = os.path.join(self.build_context, galaxy_requirements_name)
+            exists = os.path.exists(dest)
+            if not exists or not filecmp.cmp(galaxy_requirements_path, dest, shallow=False):
+                shutil.copy(galaxy_requirements_path, dest)
+
+            self.steps.extend(GalaxySteps(galaxy_requirements_name))
 
-        # There probably needs to be an intermidiate step here
-        # where we introspect the results of running the "galaxy steps"
-        # inside of the base image.
+    def prepare_pip_steps(self):
         python_req_path = self.definition.get_dependency('python')
         if python_req_path:
             shutil.copy(python_req_path, self.build_context)
+
+
+        command = [
+            self.container_runtime, "run", "--rm", self.tag, "python", "-c",
+            textwrap.dedent("""
+                import yaml
+                from ansible_builder.utils import introspect
+                print(yaml.dump(introspect(), default_flow_style=False))
```

Comment:
Just make this a method in `ansible_builder.utils`, then condense this to

```
python -c "from ansible_builder.utils import printspection; printspection()"
```

method name subject to revision.